### PR TITLE
deposit waiting for confirmation blocks

### PR DESCRIPTION
**Pull request by hackaugusto**

### CODE REVIEW

File: raiden/blockchain_events_handler.py [hunk 1]
```diff
@@ -133,10 +133,14 @@ def handle_channel_new_balance(raiden, event):
         previous_balance = previous_channel_state.our_state.contract_balance
         balance_was_zero = previous_balance == 0
 
-        newbalance_statechange = ContractReceiveChannelNewBalance(
-            channel_identifier,
+        deposit_transaction = TransactionChannelNewBalance(
```
Comment (hackaugusto) [inline raiden/blockchain_events_handler.py]: by convention *state changes* use the following prefixes:

- Receive
- ContractReceive
- Action

I used the `Transaction` prefix for this *State* following the same rationale, problem is that most state objects use the `State` **postfix**. The question is:

Should we standardize use prefixes for all these values classes, or change the `Transaction` to a postfix?

**Comment by hackaugusto** [comment]
There was two possible approaches for confirmation blocks:

- Add state to the blockchain pooler, and only dispatched events after they were confirmed
- Dispatch when the event is learned and handle confirmation inside the state machine

The second approach was chosen because it allows the logic to learn about secrets registered on-chain faster.

**Comment by LefterisJP** [comment]
I am reviewing this one now.

File: raiden/blockchain_events_handler.py (hunk 1 above)
Comment (LefterisJP) [reply raiden/blockchain_events_handler.py]: Note 1: We should have a codign document where these conventions would be noted down. 

Question: I don't get why we need two different state changes here. Isn't essentially `ContractReceiveChannelNewBalance` the deposit state change?

Here you make this new `TransactionChannelNewBalance` state change and make it a part of `ContractReceiveChannelNewBalance`. Why? What do we gain from this additional complexity? It's not immediately clear to me from reading the code. Can you maybe explain it a bit?

Comment (hackaugusto) [reply raiden/blockchain_events_handler.py]: > We should have a codign document where these conventions would be noted down

It's part of the base classes docstring:
https://github.com/raiden-network/raiden/blob/master/raiden/transfer/architecture.py#L47-L80

> Can you maybe explain it a bit?

I'm avoiding mixing state changes and state objects, `TransactionChannelNewBalance` is a `State` not a `StateChange`. My idea is to snapshot only `State` objects, not `StateChange` nor  `Event` objects.

The benefit is not huge, it's just a clear separation among these object types.

Comment (LefterisJP) [reply raiden/blockchain_events_handler.py]: I am just not sure of the benefit of adding 1 extra class but anyway. I won't interfere with your design here then.

**Comment by LefterisJP** [comment]
Will perform a rebase now.

**Comment by LefterisJP** [comment]
Rebase done. Was a bit complicated due to the other merged PRs touching the same code. Could have fucked up some things so waiting for tests.

**Comment by LefterisJP** [comment]
Tests failed. Will try to fix them.

**Comment by LefterisJP** [comment]
So @hackaugusto with all tests fixed and rebased this is approved. I leave merging to you just so you can have a final look at the diff to see if anything looks funny after rebase.

**Pull request merged**